src/gretel_trainer/benchmark/custom/datasets.py

```diff
@@ -57,3 +61,36 @@ def make_dataset(
     return CustomDataset(
         data_source=source, datatype=datatype, name=name, delimiter=delimiter
     )
+
+
+def make_dataset(
+    sources: Union[list[str], list[pd.DataFrame]],
+    *,
+    datatype: Union[str, Datatype],
+    namespace: Optional[str] = None,
+    delimiter: str = ",",
+) -> CustomDataset:
+    logger.warning(
+        "`make_dataset` is deprecated and will be removed in a future release. Please use `create_dataset` instead."
+    )
+    if len(sources) > 1:
+        raise BenchmarkException(
+            "`make_dataset` no longer supports multiple sources. Please create separate datasets using `create_dataset`."
+        )
+
+    source = sources[0]
```

Comment:
nit: What do you think about being more defensive about the type here since python will happily let you call `make_dataset("myfile.csv")` and then you'll be very confused for a while about why it's looking for a file named `"m"`.

I know the type annotation says to call this with a list, do we think that's sufficient? Since string and dataframes will both happily return something from this `sources[0]` call and then the actual error won't show up until much farther into the code, being proactive to check for a list/iterable/tuple here before grabbing `sources[0]` feels useful to me. And it's another opportunity to direct them immediately to the non-deprecated `create_dataset`.

Reply:
Yeah this is a good call, I thought about this for a moment earlier and took the lazy route, but you're right, it's worth doing. I pushed a commit that adds a pretty basic check, LMK what you think.